spec.html

```diff
@@ -268,6 +268,72 @@ <h1>AsyncIterator.from ( _O_ )</h1>
             1. Set _wrapper_.[[AsyncIterated]] to _iteratorRecord_.
             1. Return _wrapper_.
           </emu-alg>
+
+          <emu-clause id="sec-wrapforvalidasynciteratorprototype-object">
+            <h1>The %WrapForValidAsyncIteratorPrototype% Object</h1>
+            <p>The <dfn>%WrapForValidAsyncIteratorPrototype%</dfn> object:</p>
+
+            <ul>
+              <li>is an ordinary object.</li>
+              <li>has a [[Prototype]] internal slot whose value is %AsyncIterator.prototype%.</li>
+              <li>has the following properties:</li>
+            </ul>
+
+            <emu-clause id="sec-wrapforvalidasynciteratorprototype.next">
+              <h1>%WrapForValidAsyncIteratorPrototype%.next ( _value_ )</h1>
+              <emu-alg>
+                1. Let _O_ be *this* value.
+                1. Let _promiseCapability_ be ! NewPromiseCapability(%Promise%).
+                1. If Type(_O_) is not Object, or if _O_ does not have an [[AsyncIterated]] internal slot, then
```

Comment:
I think you can use RequireInternalSlot here.

Reply:
That will cause the function to throw a `TypeError` instead of returning a rejected `Promise`.

Reply:
```
1. Let _check_ be RequireObjectCoercible(_O_).
1. If _check_ is an abrupt completion, then
  1. Perform ! Call(_promiseCapability_.[[Reject]], *undefined*, << _check_.[[Value]] >>).
  1. Return _promiseCapability_.[[Promise]].
```

Reply:
IfAbruptRejectPromise may be helpful as well.